topic-operator/src/main/java/io/strimzi/operator/topic/v2/BatchingTopicController.java

```diff
@@ -961,25 +974,39 @@ private void updateStatusForException(ReconcilableTopic reconcilableTopic, Excep
             LOGGER.warnCr(reconcilableTopic.reconciliation(), "Updating status for unexpected exception", e);
             reason = e.getClass().getSimpleName();
         }
-        updateStatus(reconcilableTopic.reconciliation(), reconcilableTopic.kt(), reason, message);
+        Condition condition = new ConditionBuilder()
+            .withType("Ready")
+            .withStatus("False")
+            .withReason(reason)
+            .withMessage(message)
+            .withLastTransitionTime(StatusUtils.iso8601Now())
+            .build();
+        updateStatus(reconcilableTopic.reconciliation(), reconcilableTopic.kt(), condition);
     }
 
     private void updateStatusOk(ReconcilableTopic reconcilableTopic) {
-        updateStatus(reconcilableTopic.reconciliation(), reconcilableTopic.kt(), null, null);
+        Condition condition = new ConditionBuilder()
+            .withType(isPaused(reconcilableTopic.kt()) ? "ReconciliationPaused" : "Ready")
+            .withStatus("True")
+            .withLastTransitionTime(StatusUtils.iso8601Now())
+            .build();
+        updateStatus(reconcilableTopic.reconciliation(), reconcilableTopic.kt(), condition);
     }
 
     private void updateStatus(Reconciliation reconciliation,
                               KafkaTopic kt,
-                              String newReason,
-                              String newMessage) {
+                              Condition condition) {
         var oldStatus = kt.getStatus();
         Condition oldReadyCondition = oldStatus == null || oldStatus.getConditions() == null ? null : oldStatus.getConditions().stream().findFirst().orElse(null);
-        String newStatus = newReason != null || newMessage != null ? "False" : "True";
         if (oldStatus == null
                 || oldStatus.getObservedGeneration() != kt.getMetadata().getGeneration()
                 || oldStatus.getTopicName() == null
                 || oldReadyCondition == null
-                || isDifferentCondition(oldReadyCondition, newReason, newMessage, newStatus)) {
+                || isDifferentCondition(oldReadyCondition, condition)
+                || isPaused(kt)) {
+            long observedGeneration = oldStatus != null
+                ? !isPaused(kt) ? kt.getMetadata().getGeneration() : oldStatus.getObservedGeneration()
+                : !isPaused(kt) ? kt.getMetadata().getGeneration() : 0L;
```

Comment:
I'm not sure I follow this logic. Observed generation indicates that the operator has seen the latest change and processed it. That does not in any way change if it is paused -> the change is still processed and handled accordingly (by ignoring it in the case of paused reconciliation).

Reply:
Yes, maybe this deserve a comment. Observing the UO I found that the generation is initialized to 0 when creating a paused user (the annotation is already there on creation). This is what the last line does (oldStatus null and paused true).

Reply:
I don't remember why we do that in the UO. But yeah, comment explaining what it does might be useful.

Reply:
I think the reason is that, even in that case, we want to have a status with the observedGeneration, but you don't want to be the same as metadata.generation because the resource is not yet reconciled. I'll add a note.